components/user-mgt/org.wso2.carbon.user.mgt/src/main/java/org/wso2/carbon/user/mgt/UserAdmin.java

```diff
@@ -371,8 +377,36 @@ public void deleteRole(String roleName) throws UserAdminException {
      * @throws UserAdminException
      */
     public void updateRoleName(String roleName, String newRoleName) throws UserAdminException {
+
         try {
-            getUserAdminProxy().updateRoleName(roleName, newRoleName);
+            boolean isInternalRole = false;
+            if (roleName.startsWith(UserCoreConstants.INTERNAL_DOMAIN + UserCoreConstants.DOMAIN_SEPARATOR) ||
+                    roleName.startsWith(UserCoreConstants.APPLICATION_DOMAIN + UserCoreConstants.DOMAIN_SEPARATOR)) {
+                isInternalRole = true;
+            }
+            String internalSystemRoleName;
+            String newInternalSystemRoleName;
+            if (!isInternalRole) {
+                internalSystemRoleName = UserCoreConstants.INTERNAL_SYSTEM_ROLE_PREFIX + UserCoreUtil.extractDomainFromName(roleName)
+                        .toLowerCase() + "_" + UserCoreUtil.removeDomainFromName(roleName);
+                newInternalSystemRoleName = UserCoreConstants.INTERNAL_SYSTEM_ROLE_PREFIX + UserCoreUtil.extractDomainFromName(newRoleName)
+                        .toLowerCase() + "_" + UserCoreUtil.removeDomainFromName(newRoleName);
```

Comment:
For these also shall we use the above `getInternalSystemRoleName(String roleName)` method?